smart-contract-verifier/src/vyper/compiler.rs

```diff
@@ -0,0 +1,190 @@
+use std::path::Path;
+
+use ethers_solc::{error::SolcError, CompilerInput, CompilerOutput, Solc};
+
+use crate::compiler::{self, EvmCompilerAgent};
+
+pub struct VyperCompilerAgent {}
+
+impl VyperCompilerAgent {
+    pub fn new() -> Self {
+        VyperCompilerAgent {}
+    }
+}
+
+impl EvmCompilerAgent for VyperCompilerAgent {
+    fn compile(
+        &self,
+        path: &Path,
+        ver: &compiler::Version,
+        input: &CompilerInput,
+    ) -> Result<CompilerOutput, SolcError> {
+        let _ = ver;
+        let vyper_output: types::VyperCompilerOutput = Solc::from(path).compile_as(input)?;
+        Ok(CompilerOutput::from(vyper_output))
+    }
+}
+
+mod types {
+    use std::collections::BTreeMap;
+
+    use ethers_solc::{
+        artifacts::{Contract, Severity},
+        CompilerOutput,
+    };
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    pub struct SourceLocation {
+        pub file: String,
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct VyperError {
+        pub r#type: String,
+        pub component: String,
+        pub severity: Severity,
+        pub message: String,
+        pub formatted_message: Option<String>,
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Default)]
+    pub struct VyperCompilerOutput {
+        #[serde(default, skip_serializing_if = "Vec::is_empty")]
+        pub errors: Vec<VyperError>,
+        #[serde(default)]
+        pub contracts: BTreeMap<String, BTreeMap<String, Contract>>,
+    }
```

Comment:
What about `sources` field?

Reply:
We don't need now, but that would make it more consistent with `CompilerOutput`. Otherwise, it may result in strange outputs if someone would want to use that value but find out that it is empty for vyper even though it is presented in vyper standard output specification

Reply:
`contracts` field is just Map to AST struct. How would you suggest to map Vyper AST to Solidity AST?

Reply:
Hmm, okay... Just thought that it was a usual BTreeMap that maps source file to its string representation